src/features/DisplayControls/Allergies/components/Allergies.spec.jsx

```diff
@@ -1,47 +1,95 @@
 import { render, waitFor, screen } from "@testing-library/react";
-import axios from "axios";
 import React from "react";
-import { mockAllergiesIntolerenceResponse } from "./AllergiesTestUtils";
+import {
+  mockAllergiesIntolerenceResponse,
+  mockVisitSummaryData,
+} from "./AllergiesTestUtils";
 import Allergies from "./Allergies";
 import "@testing-library/jest-dom/extend-expect";
+import { IPDContext } from "../../../../context/IPDContext";
+
+const mockData1 = { ...mockAllergiesIntolerenceResponse.data };
+const mockUseFetchAllergiesIntolerance = jest.fn();
+
+jest.mock("../hooks/useFetchAllergiesIntolerance", () => {
+  return {
+    useFetchAllergiesIntolerance: () => mockUseFetchAllergiesIntolerance(),
+  };
+});
 
-jest.mock("axios");
 describe("Allergies", () => {
+  beforeEach(() => {
+    jest.clearAllMocks();
+    mockUseFetchAllergiesIntolerance.mockImplementation(() => ({
+      allergiesData: mockData1,
+      isLoading: false,
+    }));
+  });
+
   it("should render Allergies", () => {
-    axios.get.mockResolvedValue(mockAllergiesIntolerenceResponse);
-    render(<Allergies patientId={"__test_patient_uuid__"} />);
+    render(
+      <IPDContext.Provider
+        value={{
+          visitSummary: mockVisitSummaryData,
+        }}
+      >
+        <Allergies patientId={"__test_patient_uuid__"} />
+      </IPDContext.Provider>
+    );
+
     expect(screen.getByText("Allergen")).toBeInTheDocument();
   });
 
   it("should show data in the table", async () => {
-    axios.get.mockResolvedValue(mockAllergiesIntolerenceResponse);
-    render(<Allergies patientId={"__test_patient_uuid__"} />);
+    render(
+      <IPDContext.Provider
+        value={{
+          visitSummary: mockVisitSummaryData,
+        }}
+      >
+        <Allergies patientId={"__test_patient_uuid__"} />
+      </IPDContext.Provider>
+    );
 
     await waitFor(() => {
-      expect(screen.getByTestId(/datatable/i)).toBeInTheDocument();
+      expect(screen.getByTestId(/datatable-with-value/i)).toBeInTheDocument();
```

Comment:
is it possible to use  getByRole instead of getByTestId?

Reply:
Sure,I've replace it to getByRole